axelrod/fingerprint.py

```diff
@@ -65,168 +28,91 @@ def __init__(self, strategy, probe=axl.TitForTat):
             A class that must be descended from axelrod.Player or an instance of
             axelrod.Player.
             Default: Tit For Tat
+        step: float = 0.01
+            The separation between each Point. Smaller steps will
+            produce more Points that will be closer together.
+        progress_bar: bool = True
+            True = show a progress bar when constructing probes and running
+            tournament.
         """
-        self.strategy = strategy
-        self.probe = probe
+        self.player = create_player(strategy)
+        self._probe_class, self._probe_kwargs = get_class_and_kwargs(probe)
+        self._step = step
+        self.progress_bar = progress_bar
+        self.points = self._create_points()
+        self._probe_list = self._create_probes()
```

Comment:
So this is a bit of a change of how things are currently done. 

Having the probes created only when we choose to fingerprint was a design choice at the time. Is there a reason why you're moving it here?

From an API point of view it's different to the rest of the library where `.play()` does all the work. That is what is currently happening on master (which is why the `progress_bar` and `steps` argument is in the `fingerprint` method).

Unless there's a reason I'm missing I would prefer not to change the API too much and "just" move the static methods to private functions.